entropy/windows/Raaz/Entropy.hsc

```diff
@@ -30,18 +31,19 @@ type HCRYPTPROV = Ptr ()
 foreign import WINDOWS_CCONV unsafe "Wincrypt.h CryptGenRandom"
     c_CryptGenRandom :: HCRYPTPROV -> Word32 -> Ptr Word8 -> IO Bool
 
-foreign import WINDOWS_CCONV unsafe "Wincrypt.h CryptAcquireContext"
+foreign import WINDOWS_CCONV unsafe "Wincrypt.h CryptAcquireContextW"
     c_CryptAcquireContext :: Ptr HCRYPTPROV -> CWString -> CWString
                           -> Word32 -> Word32 -> IO Bool
 
 foreign import WINDOWS_CCONV unsafe "Wincrypt.h CryptReleaseContext"
     c_CryptReleaseContext :: HCRYPTPROV -> Word32 -> IO Bool
 
 -- | Cache the crypto context so we don't have to create it on each call.
-cryptoContext :: ForeignPtr (HCRYPTPROV)
+cryptoContext :: ForeignPtr HCRYPTPROV
```

Comment:
I think you can avoid this code. I understand that you want to cache the crypto context for performance but note that the system entropy is only used to seed the internal chacha20 based CSPRG. So calls to getEntropy is done every 64GB or so.

A similar situation occurs in the `/dev/urandom` code where we open a handle to `/dev/urandom` for every call of getEntropy. You can follow that model.